pkg/controller/accountclaim/reuse.go

```diff
@@ -77,8 +77,8 @@ func (r *ReconcileAccountClaim) finalizeAccountClaim(reqLogger logr.Logger, acco
 		return err
 	}
 
-	// Remove IAM user we'll remove the IAM user for CCS
-	if utils.AccountCRHasIAMUserIDLabel(reusedAccount) && accountClaim.Spec.BYOC {
+	// Remove IAM user we'll remove the IAM user
+	if utils.AccountCRHasIAMUserIDLabel(reusedAccount) {
```

Comment:
Not entirely in the scope of this ticket, but this `utils.AccountCRHasIAMUserIDLabel` function would probably better be moved to the Account Object itself.